qiskit/transpiler/__init__.py

```diff
@@ -40,714 +41,1132 @@
-   manner to the "physical" qubits in an actual quantum device.
+By default, qiskit will do this mapping for you.  The choice of mapping depends on the
+properties of the circuit, the particular device you are targeting, and the optimization
+level that is chosen The choice of initial layout is extremely important when:
 
-   .. image:: /source_images/mapping.png
+1. Computing the number of swap operations needed to map the input circuit onto the device
+   topology.
 
-   .. raw:: html
+2. Taking into account the noise properties of the device.
 
-      <br><br>
+The choice of `initial_layout` can mean the difference between getting a result,
+and getting nothing but noise. Due to the importance of this stage the preset pass managers
+try a few different methods to find the best layout. Typically this involves 2 stages, first
+trying to find a "perfect" layout (a layout which does not require any swap operations) and
+a heuristic pass that tries to find the best layout to use if a perfect layout can not be found.
+For the first stage there are 2 passes typically used for this:
 
-   By default, qiskit will do this mapping for you.  The choice of mapping depends on the
-   properties of the circuit, the particular device you are targeting, and the optimization
-   level that is chosen.  The basic mapping strategies are the following:
+- :class:`~.VF2Layout`: Models layout selection as a subgraph isomorphism problem and tries
+  to find a subgraph of the connectivity graph that is isomporphic to the
+  graph of 2 qubit interactions in the circuit. If more than one isomorphic mapping is found a
+  scoring heuristic is run to select the mapping which would result in the lowest average error
+  when executing the circuit.
 
-   - **Trivial layout**: Map virtual qubits to the same numbered physical qubit on the device,
-     i.e. `[0,1,2,3,4]` -> `[0,1,2,3,4]` (default in `optimization_level=0` and
-     `optimization_level=1`).
+- :class:`~.TrivialLayout`: Map virtual qubits to the same numbered physical qubit on the device,
+  i.e. ``[0,1,2,3,4]`` -> ``[0,1,2,3,4]``. This is only used in ``optimization_level=1`` to find a
+  perfect layout and if it fails :class:`~.VF2Layout` is used instead).
 
-   - **Dense layout**: Find the sub-graph of the device with same number of qubits as the circuit
-     with the greatest connectivity (default in `optimization_level=2` and `optimization_level=3`).
+Then for the heuristic pass 2 passes are used by default:
 
+- :class:`~.SabreLayout`: Used to select a layout if a perfect layout isn't found for optimization
+  levels 1, 2, and 3.
+- :class:`~.TrivialLayout`: Always used for the layout at optimization level 0
+- :class:`~.DenseLayout`: Find the sub-graph of the device with same number
+  of qubits as the circuit with the greatest connectivity. Used for
+  optimization level 1 if there are control flow operations (such as
+  :class:`~.IfElseOp`) present in the circuit.
```

Comment:
Similarly, this will likely be outdated following https://github.com/Qiskit/qiskit-terra/issues/9417 .

Reply:
Yeah, after this merges we should make a note to update this doc after when #9417 is fixed